lib/route.method.js

```diff
@@ -1,54 +1,111 @@
+var Promise = require('bluebird');
+
 var SSE = require('./sse');
+var JSONAPI_Error = require('./jsonapi-error.js');
+var sendError = require('./send-error');
+
+function methodNotAllowed(req, res) {
+    sendError(req, res, new JSONAPI_Error({status: 405}));
+}
+
+/**
+ * Create expressJS route handler.
+ *
+ * Depending on configuration it may leverage global authorizationStrategy
+ * or not.
+ * Then it delegates to apropriate handlerFunc configured during `register()`.
+ *
+ * @param routeMethod
+ * @returns {Function} expressJS route handler
+ */
+function createHandler(routeMethod) {
+    return function (request, response) {
+        var authorizationStrategy = routeMethod.options.harvester.authorizationStrategy;
+        var thatRouteHandler = this;
+        //noinspection UnnecessaryLocalVariableJS
+        var thatRouteHandlerArguments = arguments;
+
+        function forwardToHandlerFunc() {
+            routeMethod.handlerFunc.apply(thatRouteHandler, thatRouteHandlerArguments);
+        }
+
+        function isAuthorizationSupportedForMethod() {
+            return authorizationStrategy instanceof Function && routeMethod.authorizationEnabled !== false;
+        }
 
-var RouteMethod = function(options) {
-  this.options = options;
+        function handleAuthorizationResult(authorizationResult) {
+            if (null != authorizationResult && authorizationResult.then instanceof Function) {
+                authorizationResult.then(function () {
+                    Promise.method(forwardToHandlerFunc)().catch(function (error) {
+                        console.error('Route handler for', routeMethod.options.method, routeMethod.options.route,
+                            'threw exception! This should never happen!!!');
+                        sendError(request, response, error);
+                    });
+                }).catch(function () {
+                        sendError(request, response, new JSONAPI_Error({status: 403}));
+                    });
+            } else {
+                sendError(request, response, authorizationResult);
+            }
+        }
 
-  RouteMethod.prototype.register = function() {
+        if (isAuthorizationSupportedForMethod()) {
+            handleAuthorizationResult(authorizationStrategy(routeMethod.options.method, routeMethod.options.route));
+        } else {
+            forwardToHandlerFunc();
+        }
+    };
+}
+
+function RouteMethod(options) {
+    if (options.harvester == null) {
+        throw new Error('Options must include reference to harvester');
+    }
+    this.options = options;
+    this.authorizationEnabled = true;
+    return function () {
+        return this;
```

Comment:
Add little comment about doing this to deliver a function rather than an object, to meet harvester dsl requiremnts.